src/core/orderbook.js

```diff
@@ -333,16 +336,18 @@ OrderBook.prototype.destroy = function() {
  * Request orderbook entries from server
  *
  * @param {Function} callback
+ * @param {boolean} internal - internal request made on 'subscribe'
  */
 
 OrderBook.prototype.requestOffers = function(callback = function() {},
   internal = false) {
   const self = this;
 
-  if (!this._remote.isConnected()) {
+  if (!this._remote.isConnected() && !internal) {
     // do not make request if not online.
     // that requests will be queued and
     // eventually all of them will fire back
+    callback(new Error('remote is offline'));
```

Comment:
All errors passed to callbacks should be instances of RippleError (which means they can inherit from RippleError).